008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,1021 @@
+
+By calculating the length of the average reassignment period, the agents can estimate the number of days remaining on the current floor. We use the proportion of estimated days before the next reshuffling period to the duration of the treaty in order to weigh how much an agent should focus on the short term. Let $b_\text{short}$ and $b_\text{long}$ be the estimated long and short term benefit of a treaty, respectively. Also, let the estimated days remaining on the current level be given by $d_\text{current}$ and the duration of a treaty by $d_\text{treaty}$. The total benefit $b_\text{tot}$ is given by
+
+\begin{equation}
+b_\text{tot} = d\times b_\text{short} + (1-d)\times b_\text{long}    
+\label{eq:shortlongweight}
+\end{equation}
+
+where $d = d_\text{current}/d_\text{treaty}$.
+
+The following examples show how this plays out.
+
+\begin{itemize}
+    \item $d_\text{current} = 10$, $d_\text{treaty} = 8$:  $d=100\%$ (treaty expires before leaving the current floor)
+    \item $d_\text{current} = 10$, $d_\text{treaty} = 20$: $d=50\%$
+    \item $d_\text{current} = 10$, $d_\text{treaty} = 100$: $d = 10\%$
+\end{itemize}
+
+
+\subsubsection{Survival Instinct}
+
+When one is doing well, the above weighting between the long and short term benefits is sensible. However, if one is struggling to stay alive, survival instincts kick in. Consequently, the agents ignore the expected long-term utility all together when their health is on a critical level.
+
+\subsection{Implementation}
+
+How agents rate treaties closely follows the theoretical steps from above. Every day, the agent stores the amount of food received in two arrays, one for the short and one for the long term. The short-term array is reset after every reshuffling period. When a new treaty is proposed, the agent performs the following steps to conclude whether to sign this new treaty or not.
+
+\begin{enumerate}
+    \item Checks if the trust in the proposer agent is high enough
+    \item Checks if the treaty is compatible with treaties that the agent signed already
+    \item Calculates the expected short- and long-term utility is according to \Cref{eq:expectedUtility} and using the memory of food received in the past. The utility function used depends on their current social motive.
+    \item Amplifies the utility if it is negative in accordance with loss-aversion.
+    \item Estimates the amount of food it can take under the treaty, i.e. the difference between the average amount of food it received in the past and the amount of food it needs to leave in order to comply with the treaty, and calculate the utility of taking that amount of food.
+    \item Computes the estimated short-term and long-term benefit of signing the treaty as $U(\text{sign}) - U(\text{don’t sign})$.
+    \item Chooses to focus on the long-term or short-term benefit according to \Cref{eq:shortlongweight}
+    \item Signs the treaty if it’s overall benefit is positive
```

Comment:
```suggestion
    \item Check if the trust in the proposer agent is high enough.
    \item Check if the treaty is compatible with treaties that the agent signed already.
    \item Calculate the expected short- and long-term utility using the memory of food received in the past according to \Cref{eq:expectedUtility}. The utility function used depends on their current social motive.
    \item Amplify the utility if it is negative in accordance with loss-aversion.
    \item Estimate the amount of food it can take under the treaty, i.e., the difference between the average amount of food it received in the past and the amount of food it needs to leave in order to comply with the treaty, and calculate the utility of taking that amount of food.
    \item Compute the estimated short-term and long-term benefit of signing the treaty as $U(\text{sign}) - U(\text{don’t sign})$.
    \item Choose whether to focus on the long-term or short-term benefit according to \Cref{eq:shortlongweight}.
    \item Sign the treaty if its overall benefit is positive.
```